streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -304,37 +303,20 @@ public static StreamThread create(final InternalTopologyBuilder builder,
             restoreConsumer,
             userStateRestoreListener);
 
-        final ThreadCache cache = new ThreadCache(logContext, cacheSizeBytes, streamsMetrics);
-
-        final ActiveTaskCreator activeTaskCreator = new ActiveTaskCreator(
```

Comment:
I think it's actually a better idea to keep extracting the active / standby task creators out of the constructor of task-manager, since by doing that in the tests we can have mocks for those two, and also we can avoid the extra constructors you added below. Thoughts?

Also cc @abbccdda @vvcephei who reviewed / worked on the class cleanups recently.

Reply:
I guess @vvcephei refactoring did already the required work to get rid of passing around `threadProducer` and `taskProducers` references -- while I see the argument about testing and the second constructor, it still seems weird if the `StreamThread` that has nothing to do with tasks needs to create both task-creator objects -- it should be encapsulated in `TaskManager` -- maybe we can have some setters on TaskManager for testing instead of the second constructor (the corresponding members would not be `final` any longer for this case)

Reply:
Logically it is more natural to have the task creation logic inside task-manager, yes; but the tricky part is that task creation involves creating a lot of its modules like state-manager and record-collector, so if we can pass them as parameters it is easier to mock them without creating those modules; so practically speaking I think it's better to always pass in those creators as parameters to task manager so that it's easier to mock, since there's really no real disadvantage of doing so right?

Reply:
I also don't feel strong about this refactoring. In the effort of moving task creator towards `TaskManager` we are also passing in more parameters than what `TaskManager` needs as well. I would recommend we reduce the PR to address https://issues.apache.org/jira/browse/KAFKA-9676 only, since we do miss unit test coverage for task creators.